### DISCUSSION TITLE
Problem with manual installation on K3S

### DISCUSSION THREAD

**Discussion by Dzoge** [longhorn/longhorn#3760] [Q&A] [answered]
I am trying to install Longhorn on a k3s installation but some pods fail to start:

> longhorn-admission-webhook:
`Error starting longhorn admission webhook server: can't get required environment variable, env NODE_NAME wasn't set`
<details>
  <summary>Full log of 'longhorn-admission-webhook'</summary>
- Starting longhorn admission webhook server
- Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
- Waiting for caches to sync for longhorn datastore
- Caches are synced for longhorn datastore 
- Error starting longhorn admission webhook server: can't get required environment variable, env NODE_NAME wasn't se
</details>

> longhorn-driver-deployer:
`container "longhorn-driver-deployer" in pod "longhorn-driver-deployer-5bcb6c4bbc-5v74r" is waiting to start: PodInitializing`

> longhorn-manager
`container "longhorn-manager" in pod "longhorn-manager-zg4w6" is waiting to start: PodInitializing`

How can I debug this issue?

## Installation steps
I used the following steps to install Longhorn on K3S running on Rocky Linux 8. All steps succeeded.
```shell
dnf install -y --setopt=tsflags=noscripts iscsi-initiator-utils
dnf install -y nfs-utils jq
systemctl enable iscsid
systemctl start  iscsid
systemctl status iscsid

kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/v1.2.3/deploy/prerequisite/longhorn-iscsi-installation.yaml
kubectl get pod | grep longhorn-iscsi-installation
kubectl logs $(kubectl get pod -o=name | grep longhorn-iscsi-installation) -c iscsi-installation

kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/v1.2.3/deploy/prerequisite/longhorn-nfs-installation.yaml
kubectl get pod | grep longhorn-nfs-installation
kubectl logs $(kubectl get pod -o=name | grep longhorn-nfs-installation) -c nfs-installation

curl -sSfL https://raw.githubusercontent.com/longhorn/longhorn/v1.2.3/scripts/environment_check.sh | bash

kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/longhorn.yaml
```

**Comment by derekbit** [comment]
The deploy manifest in `longhorn/longhorn/deploy` is not update yet. If you'd like to try master branch, you can try to use `longhorn/longhorn-manager/deploy/install`.

**Comment by jenting** [comment, marked as answer]
We changed the branch policy recently. Therefore, the current master branch is development branch.
Please change the last script branch from master to v1.2.3 instead.

**Reply by Dzoge** [reply]
Thank you. This fixed my issue.
